core/eolearn/core/eodata_io.py

```diff
@@ -286,47 +276,25 @@ def walk_feature_type_folder(filesystem: FS, folder_path: str) -> Iterator[Tuple
             yield _remove_file_extension(path), fs.path.combine(folder_path, path)
 
 
-def walk_eopatch(
-    eopatch: EOPatch, patch_location: str, features: FeaturesSpecification
-) -> Iterator[Tuple[FeatureType, Union[str, EllipsisType], str]]:
-    """Yields tuples of (feature_type, feature_name, file_path), with file_path being the expected file path."""
-    returned_meta_features = set()
-    for ftype, fname in FeatureParser(features).get_features(eopatch):
-        ftype_path = fs.path.combine(patch_location, ftype.value)
-        if ftype.is_meta():
-            # META_INFO features are yielded separately by FeatureParser. We only yield them once with `...`,
-            # because all META_INFO is saved together
-            if eopatch[ftype] and ftype not in returned_meta_features:
-                yield (ftype, ..., ftype_path)
-                returned_meta_features.add(ftype)
-        else:
-            fname = cast(str, fname)  # name is not None for non-meta features
-            yield (ftype, fname, fs.path.combine(ftype_path, fname))
-
-
 def _check_collisions(
-    overwrite_permission: OverwritePermission,
-    files_to_save: Sequence[FeatureInfo],
-    existing_files: FilesystemDataInfo,
+    overwrite_permission: OverwritePermission, eopatch_features: List[FeatureSpec], existing_files: FilesystemDataInfo
 ):
     """Checks for possible name collisions to avoid unintentional overwriting."""
     if overwrite_permission is OverwritePermission.ADD_ONLY:
-        _check_letter_case_collisions(files_to_save, existing_files)
-        _check_add_only_permission(files_to_save, existing_files)
+        _check_letter_case_collisions(eopatch_features, existing_files)
+        _check_add_only_permission(eopatch_features, existing_files)
 
     elif platform.system() == "Windows" and overwrite_permission is OverwritePermission.OVERWRITE_FEATURES:
-        _check_letter_case_collisions(files_to_save, existing_files)
+        _check_letter_case_collisions(eopatch_features, existing_files)
 
     else:
-        _check_letter_case_collisions(files_to_save, FilesystemDataInfo())
+        _check_letter_case_collisions(eopatch_features, FilesystemDataInfo())
 
 
-def _check_add_only_permission(
-    eopatch_features: Sequence[FeatureInfo], filesystem_features: FilesystemDataInfo
-) -> None:
+def _check_add_only_permission(eopatch_features: List[FeatureSpec], filesystem_features: FilesystemDataInfo) -> None:
     """Checks that no existing feature will be overwritten."""
     unique_filesystem_features = {
-        (ftype, fname.lower()) for ftype, ftype_dict in filesystem_features.features.items() for fname in ftype_dict
+        (ftype, fname.lower()) for (ftype, fname), _ in filesystem_features.iterate_features()
```

Comment:
Could probably use the `_to_lowercase` function also here, right?

```suggestion
        _to_lowercase(*feature) for feature, _ in filesystem_features.iterate_features()
```